### DISCUSSION TITLE
Automatically download photos for offline use

### DISCUSSION THREAD

**Discussion by ridicolos** [ente-io/ente#885] [Enhancements]
I really like the idea of a option to automatically download all the photos in your libary on device and not only have them in the cloud.

So you don'thave to download the images and especially videos, if I want to show them quickly to someone. Would also come in handy if one has bad or no network.

If you have a 200MB Video which you want to show someone and have to download it first with really slow internet you are screwed.

I know that `ente` is more focused on saving disk space. But I would love to have the option either save disk space or download the photos if the device has enough storage to have them local on device and so be a bit faster when looking through your libary.

**Comment by tibert01** [comment]
I would like to have this, but also if possible, it would be interesting to downlozd the pictures to the original folder path from where they were uploaded.

For example on android : 
/DCMI/Camera for photos
/Pictures/Screenshots
And more custom folders/albums.

Not sure if it would be possible on recent versions of Android. Maybe it would require to have a full access to storage. Or to ask for each folder 1 by 1.

Also it would be interesting if no automatic sync/download, maybe allow to select all or multiple and have a download button, instead of having to go on each picture and download.

**Comment by StefanOltmann** [comment]
Yes, I need this, too.

BitTorrent Sync (now Resilio Sync) also has this option. Per default everything is downloaded on demand, but you can check a switch for specific folders or the entire thing that you want it always offline.

OneDrive & Dropbox of course have similar mechanisms.

**Comment by cof253** [comment]
This is the only extra thing I would like to add to ente! 

I've been trying to get rid of all Apple, Google, and Microsoft subscriptions and data and migrating to securer platforms. I have Proton set up for email (both personal and business) and general drive stuff, but there is no way to just view a cloud version on the mobile app without selecting it for download, waiting for it to download, and then transferring it to ANOTHER app to make edits.

**Comment by txan2** [comment]
Thanks for the neat apps. I would love to have this feature (or there is a config to not stale the caches).

**Comment by Dawnexa** [comment]
It would really be great if we can select albums or certain pictures or videos that are then kept on the device

**Comment by ghost** [comment]
I would also really like this feature so any photos/albums I have on Ente but not on my phone can be made available offline, but instead of just having them download as regular files, keep them within the Ente app and kept encrypted. Much like how you can make files in Proton Drive available offline but still encrypted.

**Comment by kenny-qin** [comment]
This is needed from a privacy perspective as well. I've encouraged someone to install Ente so that I can share photos/videos with them which is great! However, there is a certain video that they love showing other people. For this, they have manually downloaded it onto their device which is not so great because they have a lot of invasive apps on their phone that might access it. I would like to suggest to that person that they can just load it through the app instead of having a local copy but that's a hard ask because it means they have to wait a few seconds/minutes before they can show the video to others.

**Comment by antrixsamay** [comment]
This is a must needed feature as it increases usability of app for users in no network scenarios

**Comment by bradall** [comment]
Would love this as a feature, specifically to see all photos in poor network conditions.
